run/cardano2john.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+#
+# Export Cardano's legacy secret.key to John The Ripper format
+#
+# This software is Copyright (c) 2022, Pal Dorogi (ilap) <pal dot dorogi at gmail.com>
+# and it is hereby released to the general public under the following terms:
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted.
+#
+# Requireements: pip3 at OS level
+#
+# Installation:
+# $ pip3 install argparse binascii cbor2
+#
+
+import sys
+import os
+
+import argparse
+import binascii
+from cbor2 import load
+
+
+def esk2john(user_key):
+    return "$cardano$1$" + binascii.hexlify(user_key[0]).decode("ascii")
+
+
+def export_to_john(filename):
+
+    try:
+        with open(filename, "rb") as fp:
+            # cbor[0]: _usVssMaybe, not interesting
+            # cbor[1]: _usPrimKey,  not interesting
+            # cbor[2]: _usKeys, interested only the esk part, atm, [[esk, passwordHash],...,]
+            # cbor[3]: _usWalletSet, interested if there is any, but only in its [0]
+            # i.e.: _wusRootKey which is an esk
+            cbor = load(fp)
+
+            # Go through all user's keys a.k.a _usKeys and the the Wallet set.
+            keys = cbor[2] + cbor[3]
+            keys_len = len(keys)
+
+            if keys_len == 0:
+                print("No keys found in the keystore file.")
+            else:
+                out_file = "cardano.esks"
+                try:
+                    with open(out_file, "w") as fp:
```

Comment:
This is inconsistent with other 2john scripts, which output to stdout (and expect it to be redirected). I think we should be consistent.

Reply:
Yeah, I refactored the code completely.